wfcGen.py

```diff
@@ -0,0 +1,287 @@
+                    cells[str(len(cells.keys())-1)]["img"] = cropped[ins][y][x]
+
+    for y in range(lcy):
+        for x in range(lcx):
+            for key in cells.keys():
+                if cells[key]["img"] == cropped[ins][y][x]:
+                    Gkey = key
+            for key in cells.keys():
+                for i in range(len(cells[key]["nbg"]["0"])):
+                    if cells[key]["nbg"]["0"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["0"][i] = Gkey
+                for i in range(len(cells[key]["nbg"]["1"])):
+                    if cells[key]["nbg"]["1"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["1"][i] = Gkey
+                for i in range(len(cells[key]["nbg"]["2"])):
+                    if cells[key]["nbg"]["2"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["2"][i] = Gkey
+                for i in range(len(cells[key]["nbg"]["3"])):
+                    if cells[key]["nbg"]["3"][i] == cropped[ins][y][x]:
+                        cells[key]["nbg"]["3"][i] = Gkey
+
+    print(cells)
+
+def intersection(mass1, mass2):
+    return list(set(mass1) & set(mass2))
+
+class Cell:
+    def __init__(self, gp):
+        self.img = 0
+        self.id = -1
+        self.nbg = []
+        self.gp = gp
+        self.collapsed = False
+        self.entropy = []
+        self.anyFlag = True
+    
+    def collapse(self):
+        '''
+        COLAPS YEAHHH
+        '''
+        if len(self.entropy) > 0:
+            self.collapsed = True
+            self.id = random.choice(self.entropy)
+            self.entropy = [self.id]
+            pick = cells[self.id]
+            self.nbg = pick["nbg"]
+            self.img = pick["img"]
+    
+    def setCell(self, idc):
+        self.collapsed = True
+        self.id = idc
+        self.entropy = [idc]
+        pick = cells[str(idc)]
+        self.nbg = pick["nbg"]
+        self.img = pick["img"]            
+    
+    def loadEntropy(self, grid):
+        if not self.collapsed:
+            self.anyFlag = True
+            if grid[self.gp[1]-1][self.gp[0]].collapsed:
+                self.entropy = grid[self.gp[1]-1][self.gp[0]].nbg["3"]
+                self.anyFlag = False
+            elif grid[self.gp[1]][self.gp[0]-1].collapsed:
+                self.entropy = grid[self.gp[1]][self.gp[0]-1].nbg["2"]
+                self.anyFlag = False
+            elif (self.gp[1]+1)<len(grid):
+                if grid[self.gp[1]+1][self.gp[0]].collapsed:
+                    self.entropy = grid[self.gp[1]+1][self.gp[0]].nbg["1"]
+                    self.anyFlag = False
+                elif (self.gp[0]+1)<len(grid[0]):
+                    if grid[self.gp[1]][self.gp[0]+1].collapsed:
+                        self.entropy = grid[self.gp[1]][self.gp[0]+1].nbg["0"]
+                        self.anyFlag = False
+                    else:
+                        return

```

Comment:
Again, hard to read, a lot of duplication. Please, fix here as well as in other places